### DISCUSSION TITLE
Layout file created automatically inside dynamic routes when having only `page.tsx`

### DISCUSSION THREAD

**Discussion by mupinnn** [vercel/next.js#67752] [Help]
### Summary

Is it really necessary to create "bypass" layout in dynamic routes? I got this warning `Your page app/[slug]/page.tsx did not have a root layout. We created app/[slug]/layout.tsx for you.` in my terminal and Next.js automatically created the layout file.

```tsx
export const metadata = {
  title: 'Next.js',
  description: 'Generated by Next.js',
}

export default function RootLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html lang="en">
      <body>{children}</body>
    </html>
  )
}
```

Then, I simplify the generated layout to only bypass because I only need the root layout (`app/layout.tsx`).

```tsx
export default function RootLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return children;
}
```

### Additional information

```js
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Fri Mar 29 23:14:13 UTC 2024
Binaries:
  Node: 20.12.2
  npm: 10.5.0
  Yarn: N/A
  pnpm: 8.15.5
Relevant Packages:
  next: 14.1.0
  eslint-config-next: 14.1.0
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

**Comment by icyJoseph** [comment]
What you need is a root layout. It doesn't matter that you are using dynamic routes, or static routes. 

See, https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required:

> This layout is required and must contain html and body tags, allowing you to modify the initial HTML returned from the server.

In some scenarios, when a page fetches rsc, it will unmount the dynamic route components, and remount them, and the closest layout is a place where components won't be remounted.

**Reply by mupinnn** [reply]
Yes, I did put the root layout; but when I add the nested route with only page file, I'm expecting it will use the root layout and it works well. I only wondering why Next.js generate the layout file on my nested route.

Before:
```
- app
- - page.tsx
- - layout.sx
- - [slug]
- - - - page.tsx
```
Suddenly, this warning appear: "Your page app/[slug]/page.tsx did not have a root layout. We created app/[slug]/layout.tsx for you."
After that:
```
- app
- - page.tsx
- - layout.sx
- - [slug]
- - - - page.tsx
- - - - layout.tsx
```

**Reply by icyJoseph** [reply]
Aha, that's not how I read your first post. That shouldn't be necessary though. See here, https://stackblitz.com/edit/nextjs-zmmxsr?file=package.json

It uses 14.1.0, and the file structure is similar to yours.

<img width="471" alt="Screenshot 2024-07-16 at 09 56 21" src="https://github.com/user-attachments/assets/2f9b334b-689e-4ae4-9438-58fe4beb927b">

**Reply by mupinnn** [reply]
Sorry for the confusion. Yeah, I think that not necessary too because it will use the outer layout until reached the root.

Next.js keep generating the file for me if I delete it 😅

**Reply by icyJoseph** [reply]
That's really odd 😅 ~ What next.js version do you have? your initial post says 14.1.0, but is that correct? Maybe it is a linux thing? or there's a typo somewhere? really strange

**Reply by mupinnn** [reply]
Yes, that's correct version. Nah idk, still wondering why hahaha. Here is the file that generated if you want to check out https://github.com/mupinnn/mupin.dev/blob/main/apps/memoir/app/%5Bslug%5D/layout.tsx

**Reply by icyJoseph** [reply]
I can't reproduce that though haha, see here: https://stackblitz.com/edit/github-tde8zt?file=app%2F[slug]%2Fpage.tsx

I modified your project so that it could run in a Stackblitz environment, and it doesn't happen. I imagine it is an issue with your project setup though...

**Reply by mupinnn** [reply]
It's okaay, I might just ignore it since it doesn't bother me at all. Just curious.

Yeah, probably will check my setup. Appreciate it, Thank you so much!
